### PR TITLE
Add more context to recovered panics

### PR DESCRIPTION
**Pull request by e-dard**
###### Required for all non-trivial PRs
- [ ] Rebased/mergable
- [ ] Tests pass

We need more context when panics are recovered.

### CODE REVIEW

**Comment by dgnorton** [comment]
Do we want the whole stack?

**Comment by e-dard** [comment]
@dgnorton I don't think in these cases we can have too much.

**Comment by corylanou** [comment]
This will likely break any log won't it?  Multiple lines?  Can you show an example of the output?

**Comment by dgnorton** [comment]
:+1:

**Comment by jwilder** [comment]
Should we add more context similar to: https://github.com/hashicorp/terraform/blob/d1ac7d3b2e1be2f32092275340f33a2d73f7b31a/panic.go#L15

File: services/httpd/handler.go [hunk 1]
```diff
@@ -929,7 +930,7 @@ func (h *Handler) recovery(inner http.Handler, name string) http.Handler {
 		defer func() {
 			if err := recover(); err != nil {
 				logLine := buildLogLine(l, r, start)
-				logLine = fmt.Sprintf(`%s [panic:%s]`, logLine, err)
```
Comment (benbjohnson) [inline services/httpd/handler.go]: I'm not sure we even need the wrapping `%s [panic ]`. I would vote to just dump `debug.Stack()` as-is.

**Comment by e-dard** [comment]
@benbjohnson the stack won't contain the reason for the panic so I guess we need to keep the explicit logging of `err`.

**Comment by e-dard** [comment]
I've removed the extra `\n\n`. @corylanou example output would be:

```
[monitor] 2016/05/20 17:49:05 'network' registered for diagnostics monitoring
[monitor] 2016/05/20 17:49:05 'system' registered for diagnostics monitoring
[monitor] 2016/05/20 17:49:05 Storing statistics in database '_internal' retention policy 'monitor', at interval 10s
[httpd] 2016/05/20 17:49:05 ::1 - - [20/May/2016:17:49:05 +0100] POST /cluster/join HTTP/1.1 204 0 - Go-http-client/1.1 c3abb6ce-1eaa-11e6-8001-000000000000 144.164384ms
[httpd] 2016/05/20 17:49:15 ::1 - - [20/May/2016:17:49:15 +0100] POST /query?chunked=true&db=db34&epoch=ns&q=create+database+db34 HTTP/1.1 200 46 - InfluxDBShell/unknown c941a0db-1eaa-11e6-8002-000000000000 7.171779ms
[httpd] 2016/05/20 17:49:16 ::1 - - [20/May/2016:17:49:16 +0100] POST /query?db=&epoch=ns&q=SHOW+DATABASES HTTP/1.1 200 105 - InfluxDBShell/unknown ca12fb3e-1eaa-11e6-8003-000000000000 1.072362ms
[httpd] 2016/05/20 17:49:17 ::1 - - [20/May/2016:17:49:17 +0100] POST /write?consistency=all&db=db34&precision=ns&rp= HTTP/1.1 200 23 - InfluxDBShell/unknown cac67e56-1eaa-11e6-8004-000000000000 6.590228ms [panic:runtime error: invalid memory address or nil pointer dereference]

goroutine 73 [running]:
runtime/debug.Stack(0x0, 0x0, 0x0)
	/usr/local/share/go/src/runtime/debug/stack.go:24 +0x80
github.com/influxdata/influxdb/services/httpd.(*Handler).recovery.func1.1(0xc8200b8900, 0xc82016e2a0, 0xeced1370d, 0xc827b011f6, 0xcef4e0, 0xc820166090)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:933 +0xb7
panic(0x75bee0, 0xc820014100)
	/usr/local/share/go/src/runtime/panic.go:443 +0x4e9
github.com/influxdata/plutonium/cluster.(*PointsWriter).MapShards(0xc82014cb00, 0xc8202ced28, 0x4, 0x0, 0x0)
	/Users/edd/Development/golang/src/github.com/influxdata/plutonium/cluster/points_writer.go:208 +0x713
github.com/influxdata/plutonium/cluster.(*PointsWriter).WritePoints(0xc82014cb00, 0xc8201c629f, 0x4, 0xc82035aef0, 0x7, 0x3, 0xc8201841f0, 0x1, 0x1, 0x0, ...)
	/Users/edd/Development/golang/src/github.com/influxdata/plutonium/cluster/points_writer.go:233 +0x1ff
github.com/influxdata/influxdb/services/httpd.(*Handler).serveWrite(0xc820166090, 0x1756260, 0xc8200b8980, 0xc82016e2a0, 0x0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:548 +0x1510
github.com/influxdata/influxdb/services/httpd.(*Handler).(github.com/influxdata/influxdb/services/httpd.serveWrite)-fm(0x1756260, 0xc8200b8980, 0xc82016e2a0, 0x0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:120 +0x48
github.com/influxdata/influxdb/services/httpd.authenticate.func1(0x1756260, 0xc8200b8980, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:724 +0x73
net/http.HandlerFunc.ServeHTTP(0xc8201a2bc0, 0x1756260, 0xc8200b8980, 0xc82016e2a0)
	/usr/local/share/go/src/net/http/server.go:1618 +0x3a
github.com/influxdata/influxdb/services/httpd.gzipFilter.func1(0x1756188, 0xc8200b8920, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:854 +0x25d
net/http.HandlerFunc.ServeHTTP(0xc8201a2be0, 0x1756188, 0xc8200b8920, 0xc82016e2a0)
	/usr/local/share/go/src/net/http/server.go:1618 +0x3a
github.com/influxdata/influxdb/services/httpd.versionHeader.func1(0x1756188, 0xc8200b8920, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:863 +0xc0
net/http.HandlerFunc.ServeHTTP(0xc8201a2c00, 0x1756188, 0xc8200b8920, 0xc82016e2a0)
	/usr/local/share/go/src/net/http/server.go:1618 +0x3a
github.com/influxdata/influxdb/services/httpd.cors.func1(0x1756188, 0xc8200b8920, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:901 +0x406
net/http.HandlerFunc.ServeHTTP(0xc8201a2c20, 0x1756188, 0xc8200b8920, 0xc82016e2a0)
	/usr/local/share/go/src/net/http/server.go:1618 +0x3a
github.com/influxdata/influxdb/services/httpd.requestID.func1(0x1756188, 0xc8200b8920, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:911 +0x15d
net/http.HandlerFunc.ServeHTTP(0xc8201a2c40, 0x1756188, 0xc8200b8920, 0xc82016e2a0)
	/usr/local/share/go/src/net/http/server.go:1618 +0x3a
github.com/influxdata/influxdb/services/httpd.(*Handler).logging.func1(0x1756188, 0xc8200b8900, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:919 +0x10b
net/http.HandlerFunc.ServeHTTP(0xc8201a2c60, 0x1756188, 0xc8200b8900, 0xc82016e2a0)
	/usr/local/share/go/src/net/http/server.go:1618 +0x3a
github.com/influxdata/influxdb/services/httpd.(*Handler).recovery.func1(0x1756150, 0xc8200601a0, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:938 +0x168
net/http.HandlerFunc.ServeHTTP(0xc8201a2c80, 0x1756150, 0xc8200601a0, 0xc82016e2a0)
	/usr/local/share/go/src/net/http/server.go:1618 +0x3a
github.com/bmizerany/pat.(*PatternServeMux).ServeHTTP(0xc820150020, 0x1756150, 0xc8200601a0, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/bmizerany/pat/mux.go:109 +0x244
github.com/influxdata/influxdb/services/httpd.(*Handler).ServeHTTP(0xc820166090, 0x1756150, 0xc8200601a0, 0xc82016e2a0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/services/httpd/handler.go:200 +0x45f
net/http.serverHandler.ServeHTTP(0xc8201d2100, 0x1756150, 0xc8200601a0, 0xc82016e2a0)
	/usr/local/share/go/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc820088180)
	/usr/local/share/go/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
	/usr/local/share/go/src/net/http/server.go:2137 +0x44e

... log continues
```

**Comment by benbjohnson** [comment]
@e-dard Good point. 👍

**Comment by jwilder** [comment]
@e-dard There is a query recover handler as well.  We should update that too.

**Comment by jwilder** [comment]
https://github.com/influxdata/influxdb/blob/1dcc9f1e6362a0e553e791a1f05ab5658760f2c5/influxql/query_executor.go#L296

**Comment by jwilder** [comment]
👍

**Pull request merged**